tools/build-tool/code/tooldebian.pas

```diff
@@ -0,0 +1,192 @@
+  PathToIconFileLocal := StringReplace(PathToIconFileUnix, '/', PathDelim, [rfReplaceAll]);
+  if Manifest.Icons.FindExtension(['.xpm']) <> '' then
+    CheckCopyFile(Manifest.Icons.FindExtension(['.xpm']), PackageFolder + PathToIconFileLocal)
+  else
+  begin
+    // using ImageMagic - FPWriteXPM first doesn't properly write alpha channel, second uses palette char size = 2 which is not a good idea for an icon
+    //RunCommandSimple(FindExe('convert'), [Manifest.Icons.FindExtension(['.png']), PackageFolder + PathToIconFileLocal]);
+    if not RunCommand('/bin/convert', [Manifest.Icons.FindExtension(['.png']), PackageFolder + PathToIconFileLocal], OutString) then
+      WriteLn('ImageMagick failed.');
+    WriteLn(OutString);
+  end;
+
+  // Create menu item for the game
+
+  CreateDir(ShareDir + PathDelim + 'menu');
+  TextWriter := TTextWriter.Create(ShareDir + PathDelim + 'menu' + PathDelim + Manifest.Name);
+  TextWriter.Write(
+    '?package(' + Manifest.Name + '): \' + NL +
+    'needs="X11" \' + NL +
+    'section="' + Manifest.DebianSection + '" \' + NL +
+    'title="' + Manifest.Caption + '" \' + NL +
+    'command="bash -c ''cd ' + PathToExecutableUnix + ' && ./' + Manifest.ExecutableName + '''" \' + NL +
+    'icon="' + PathToIconFileUnix +'"'
+    );
+  FreeAndNil(TextWriter);
+
+  CreateDir(ShareDir + PathDelim + 'applications');
+  TextWriter := TTextWriter.Create(ShareDir + PathDelim + 'applications' + PathDelim + Manifest.ExecutableName + '.desktop');
+  TextWriter.Write(
+    '[Desktop Entry]' + NL +
+    'Version=' + Manifest.Version.DisplayValue + NL +
+    'Terminal=false' + NL +
+    'Exec=bash -c ''cd ' + PathToExecutableUnix + ' && ./' + Manifest.ExecutableName + '''' + NL +
+    'Icon=' + PathToIconFileUnix + NL +
+    'Type=Application' + NL +
+    'Categories=' + Manifest.DebianCategories + NL +
+    'Name=' + Manifest.Caption + NL +
+    'Comment=' + Manifest.DebianComment
+    );
+  FreeAndNil(TextWriter);
+
+  CreateDir(PackageFolder + PathDelim + 'DEBIAN');
+  TextWriter := TTextWriter.Create(PackageFolder + PathDelim + 'DEBIAN' + PathDelim + 'control');
+  TextWriter.Write(
+    'Package: ' + Manifest.Name + NL +
+    'Version: ' + Manifest.Version.DisplayValue + NL +
+    'Section: ' + Manifest.DebianInstallFolder + NL +
+    'Priority: optional' + NL +
+    'Architecture: ' + CpuToArchitectureString(Cpu) + NL +
+    'Maintainer: ' + Manifest.Author + NL +
+    'Installed-Size: ' + IntToStr(BinariesSize div 1024) + NL +
+    'Depends: libopenal1, libpng16-16, zlib1g, libvorbis0a, libvorbisfile3, libfreetype6, libgl1-mesa-dri, libgtk2.0-0' + NL +
+    'Description: ' + Manifest.Caption + NL +
+    ' ' + Manifest.DebianComment + NL //final new line
+    );
+  FreeAndNil(TextWriter);
+
+  // Post-installation running - assign executable permissions
+
+  TextWriter := TTextWriter.Create(PackageFolder + PathDelim + 'DEBIAN' + PathDelim + 'postinst');
```

Comment:
To simply create a file from a string contents, use `StringToFile`. It is simpler to use than creating / destroying `TTextWriter` instance.

Reply:
Fixed in 492b70590837280d3642fd17b669ed4cf5b84465 and fc48d68471b1e5639df5260b441a4e25d0ceafd7